csa.rs

```diff
@@ -0,0 +1,134 @@
+use std::io;
+
+const MAX_STATIONS: usize = 100000;
+
+#[derive(Debug)]
+struct Connection {
+    departure_station: usize,
+    arrival_station: usize,
```

Comment:
using `u32` seems much better as it is read as u32 and u32 is more effective. That'll also avoid the uglies `std::u32::MAX as usize`